### PR TITLE
[ELY-1025] - Provide a Bearer Token HTTP Authentication Mechanism

### PR DESCRIPTION
**Pull request by pedroigor**

### CODE REVIEW

**Comment by pedroigor** [comment]
Marking as critical because QA is expecting this mechanism to better test ```TokenSecurityRealm```. It will also improve our support for OAuth2 and OpenID Connect.

File: src/main/java/org/wildfly/security/http/impl/BearerTokenAuthenticationMechanism.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+ * <pre>
+ *      GET /resource-server HTTP/1.1
+ *      Host: elytron.org
+ *      Authorization: Bearer hgTasdMNMMAsii
+ * </pre>
+ *
+ *
+ * @author <a href="mailto:psilva@redhat.com">Pedro Igor</a>
+ */
+public class BearerTokenAuthenticationMechanism implements HttpServerAuthenticationMechanism {
```
Comment (darranl) [inline src/main/java/org/wildfly/security/http/impl/BearerTokenAuthenticationMechanism.java]: Should be default visibility here.

**Comment by darranl** [comment]
And looks like another conflict in there to be resolved.

**Comment by pedroigor** [comment]
Rebased.

**Pull request merged**